File: java/spring-boot/src/main/java/org/graphqlize/java/springboot/GraphQLizeResolverProvider.java
package org.graphqlize.java.springboot;

import org.graphqlize.java.GraphQLizeResolver;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

import javax.sql.DataSource;

@Component
public class GraphQLizeResolverProvider {
  private final DataSource dataSource;
  private final GraphQLizeResolver graphQLizeResolver;

  public GraphQLizeResolverProvider(DataSource dataSource) {
    this.dataSource = dataSource;
    graphQLizeResolver = new GraphQLizeResolver(dataSource);
  }

  @Bean
  public GraphQLizeResolver graphQLizeResolver() {
    return this.graphQLizeResolver;
  }
}

File: kotlin/spring-boot/src/main/kotlin/org/graphqlize/kotlinspringbootdemo/GraphQLizeResolverProvider.kt
package org.graphqlize.kotlinspringbootdemo

import org.graphqlize.java.GraphQLizeResolver
import org.springframework.context.annotation.Bean
import org.springframework.stereotype.Component
import javax.sql.DataSource

@Component
class GraphQLizeResolverProvider(dataSource : DataSource) {
    val graphQLizeResolver = GraphQLizeResolver(dataSource)
    @Bean
    fun graphQLizeResolver() = graphQLizeResolver
}


